src/robusta/core/playbooks/internal/discovery_events.py

```diff
@@ -17,3 +18,59 @@ def cluster_discovery_updates(event: KubernetesAnyChangeEvent):
                 namespace=event.obj.metadata.namespace,
             )
         )
+
+
+@action
+def event_history(event: ExecutionBaseEvent):
+    """
+    Creates findings for the all the past events for any object that has had at one point a warning event
+    """
+    reported_obj_history_list = []
+    warning_events = get_events_list(event_type="Warning")
+    for warning_event in warning_events.items:
+        warning_event_key = gen_object_key(warning_event.involvedObject)
+        if warning_event_key in reported_obj_history_list:
+            # if there were multiple warnings on the same object we dont want the history pulled multiple times
+            continue
+        finding = create_event_finding(warning_event)
+        events_table = get_resource_events_table(
+            "Resource events",
+            warning_event.involvedObject.kind,
+            warning_event.involvedObject.name,
+            warning_event.involvedObject.namespace
+        )
+        if events_table:
+            finding.add_enrichment([events_table])
+        event.add_finding(finding)
+        reported_obj_history_list.append(warning_event_key)
+
+
+def create_event_finding(event: Event):
+    """
+    Create finding based on the kubernetes event
+    """
+    k8s_obj = event.involvedObject
+
+    finding = Finding(
+        title=f"{event.reason} {event.type} for {k8s_obj.kind} {k8s_obj.namespace}/{k8s_obj.name}",
+        description=event.message,
+        source=FindingSource.KUBERNETES_API_SERVER,
+        severity=FindingSeverity.MEDIUM,
+        finding_type=FindingType.ISSUE,
+        aggregation_key=f"Kubernetes {event.type} Event",
+        subject=FindingSubject(
+            k8s_obj.name,
+            FindingSubjectType.from_kind(k8s_obj.kind.lower()),
+            k8s_obj.namespace,
+        ),
+        creation_date= get_event_timestamp(event)
+    )
+    finding.service_key = TopServiceResolver.guess_service_key(name=k8s_obj.name, namespace=k8s_obj.namespace)
+    # non-running services won't be in top service resolver
+    if not finding.service_key:
+        finding.service_key = gen_object_key(k8s_obj)
```

Comment:
Service key shouldn't have the pod name.
It can be empty though.
I would either leave it empty, or remove the whole finding

Reply:
for services that are no longer running and jobs it would mean we loose all the data, this way we would be able to connect relevant services in the future.
Additionally these are simmilar parameters passed to and displayed from ServiceInfo to make the service key for events. So i would like to understand how its different here?
`            
          TopServiceResolver.add_cached_service(
            ServiceInfo(
                name=event.obj.metadata.name,
                service_type=event.obj.kind,
                namespace=event.obj.metadata.namespace,
            )
        )
`

Reply:
This is just for configuration entities, 
`event.obj.kind in ["Deployment", "ReplicaSet", "DaemonSet", "StatefulSet"]`
Not for jobs/pods which have random names.
The service_key field refers to a top level entity, which is a microservice/Application

Reply:
I would just add it without one.
We already have such findings